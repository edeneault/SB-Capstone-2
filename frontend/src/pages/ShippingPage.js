import React, { useState } from "react";
import { Container, Form, Button } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import FormContainer from "../components/FormContainer";
import CheckoutSteps from "../components/CheckoutSteps";
import { saveShippingAddress } from "../actions/cartActions";

const ShippingPage = ({ history }) => {
  const cart = useSelector((state) => state.cart);
  const { shippingAddress } = cart;

  const [address, setAddress] = useState(shippingAddress.address);
  const [city, setCity] = useState(shippingAddress.city);
  const [postalCode, setPostalCode] = useState(shippingAddress.postalCode);
  const [country, setCountry] = useState(shippingAddress.country);

  const dispatch = useDispatch();

  const submitHandler = (e) => {
    e.preventDefault();
    dispatch(saveShippingAddress({ address, city, postalCode, country }));
    history.push("/payment");
  };

  return (
    <Container fluid className='px-5 pt-3 fade-in'>
      <FormContainer>
        <CheckoutSteps step1 step2 />
        <h1>Shipping</h1>
        <Form onSubmit={submitHandler}>
          <Form.Group controlId='Address'>
            <Form.Label>Address</Form.Label>
            <Form.Control
              type='text'
              placeholder='Enter Address'
              value={address}
              onChange={(e) => setAddress(e.target.value)}
              required
            ></Form.Control>
          </Form.Group>

          <Form.Group controlId='City'>
            <Form.Label>City</Form.Label>
            <Form.Control
              type='text'
              placeholder='Enter City'
              value={city}
              onChange={(e) => setCity(e.target.value)}
              required
            ></Form.Control>
          </Form.Group>

          <Form.Group controlId='PostalCode'>
            <Form.Label>PostalCode</Form.Label>
            <Form.Control
              type='text'
              placeholder='Enter PostalCode'
              value={postalCode}
              onChange={(e) => setPostalCode(e.target.value)}
              required
            ></Form.Control>
          </Form.Group>

          <Form.Group controlId='country'>
            <Form.Label>country</Form.Label>
            <Form.Control
              type='text'
              placeholder='Enter country'
              value={country}
              onChange={(e) => setCountry(e.target.value)}
              required
            ></Form.Control>
          </Form.Group>

          <Button type='submit' variant='primary' className=' rounded my-2'>
            Continue
          </Button>
        </Form>
      </FormContainer>
    </Container>
  );
};

export default ShippingPage;
