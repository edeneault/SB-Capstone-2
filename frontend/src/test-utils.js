import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
