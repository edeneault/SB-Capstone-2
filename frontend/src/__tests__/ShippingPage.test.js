import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";
import ShippingPage from "../pages/ShippingPage";
import { createMemoryHistory } from "history";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  const history = createMemoryHistory();

  test("renders Shipping without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <ShippingPage history={history} />
      </Provider>,
      { initialRoutes: [`/shipping`] },
    );
    // screen.debug();
    expect(getByText(/address/i)).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <ShippingPage history={history} />
      </Provider>,
      { initialRoutes: [`/shipping`] },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
