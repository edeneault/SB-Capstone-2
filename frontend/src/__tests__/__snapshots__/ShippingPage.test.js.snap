// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-5 pt-3 container-fluid"
  >
    <div
      class="container"
    >
      <div
        class="justify-content-md-center row"
      >
        <div
          class="col-md-6 col-12"
        >
          <div
            class="row"
          >
            <div
              class="col-12"
            >
              <div
                class="justify-content-md-center mb-4 nav"
              >
                <div
                  class="mx-1 px-1 nav-item"
                >
                  <a
                    class="nav-link"
                    data-rb-event-key="/login"
                    href="/login"
                  >
                    Sign In
                  </a>
                </div>
                <div
                  class="mx-1 px-1 nav-item"
                >
                  <a
                    class="active nav-link"
                    data-rb-event-key="/shipping"
                    href="/shipping"
                  >
                    Shipping
                  </a>
                </div>
                <div
                  class="mx-1 px-1 nav-item"
                >
                  <a
                    aria-disabled="true"
                    class="nav-link disabled"
                    href="#"
                    role="button"
                    tabindex="-1"
                  >
                    Payment
                  </a>
                </div>
                <div
                  class="mx-1 px-1 nav-item"
                >
                  <a
                    aria-disabled="true"
                    class="nav-link disabled"
                    href="#"
                    role="button"
                    tabindex="-1"
                  >
                    Place Order
                  </a>
                </div>
              </div>
            </div>
          </div>
          <h1>
            Shipping
          </h1>
          <form
            class=""
          >
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="Address"
              >
                Address
              </label>
              <input
                class="form-control"
                id="Address"
                placeholder="Enter Address"
                required=""
                type="text"
                value="9620 W. Russell Rd   APT 1025"
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="City"
              >
                City
              </label>
              <input
                class="form-control"
                id="City"
                placeholder="Enter City"
                required=""
                type="text"
                value="Las Vegas"
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="PostalCode"
              >
                PostalCode
              </label>
              <input
                class="form-control"
                id="PostalCode"
                placeholder="Enter PostalCode"
                required=""
                type="text"
                value="89148"
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="country"
              >
                country
              </label>
              <input
                class="form-control"
                id="country"
                placeholder="Enter country"
                required=""
                type="text"
                value="United States"
              />
            </div>
            <button
              class=" rounded my-2 btn btn-primary"
              type="submit"
            >
              Continue
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
